Sources/TuistDependencies/SwiftPackageManager/SwiftPackageManagerInteractor.swift

```diff
@@ -0,0 +1,190 @@
+    }
+
+    // MARK: - Installation
+
+    private func prepareForInstallation(pathsProvider: SwiftPackageManagerPathsProvider, dependencies: SwiftPackageManagerDependencies) throws {
+        // copy `.build` directory from previous run if exist
+        if fileHandler.exists(pathsProvider.destinationSwiftPackageManagerBuildDirectory) {
+            try copyDirectory(
+                from: pathsProvider.destinationSwiftPackageManagerBuildDirectory,
+                to: pathsProvider.temporarySwiftPackageManagerBuildDirectory
+            )
+        }
+
+        // copy `Package.resolved` directory from previous run if exist
+        if fileHandler.exists(pathsProvider.destinationPackageResolvedPath) {
+            try copyDirectory(
+                from: pathsProvider.destinationPackageResolvedPath,
+                to: pathsProvider.temporaryPackageResolvedPath
+            )
+        }
+
+        // create `Package.swift`
+        let packageManifestContent = dependencies.stringValue()
+        let packageManifestPath = pathsProvider.temporaryDirectoryPath.appending(component: "Package.swift")
+        try fileHandler.write(packageManifestContent, path: packageManifestPath, atomically: true)
+
+        // log
+        logger.info("Package.swift:", metadata: .subsection)
+        logger.info("\(packageManifestContent)")
+    }
+
+    private func buildCommand(packagePath: AbsolutePath) -> [String] {
+        let command = ["swift", "package", "--package-path", "\(packagePath.pathString)", "resolve"]
+
+        // log
+        logger.info("Command:", metadata: .subsection)
+        logger.info("\(command.joined(separator: " "))")
+
+        return command
+    }
+
+    private func postInstallationActions(pathsProvider: SwiftPackageManagerPathsProvider) throws {
+        // validation
+        guard fileHandler.exists(pathsProvider.temporaryPackageResolvedPath) else {
+            throw SwiftPackageManagerInteractorError.packageResolvedNotFound
+        }
+        guard fileHandler.exists(pathsProvider.temporarySwiftPackageManagerBuildDirectory) else {
+            throw SwiftPackageManagerInteractorError.buildDirectoryNotFound
+        }
+
+        // save `Package.resolved`
+        try copyFile(
+            from: pathsProvider.temporaryPackageResolvedPath,
+            to: pathsProvider.destinationPackageResolvedPath
+        )
+
+        // save `.build` direcotry
+        try copyDirectory(
+            from: pathsProvider.temporarySwiftPackageManagerBuildDirectory,
+            to: pathsProvider.destinationSwiftPackageManagerBuildDirectory
+        )
+    }
+
+    // MARK: - Helpers
+
+    private func copyFile(from fromPath: AbsolutePath, to toPath: AbsolutePath) throws {
+        try fileHandler.createFolder(toPath.removingLastComponent())
+
+        if fileHandler.exists(toPath) {
+            try fileHandler.replace(toPath, with: fromPath)
+        } else {
+            try fileHandler.copy(from: fromPath, to: toPath)
```

Comment:
does copy fail if a destination exists? If not, I'd merge `copyFile` and `copyDirectory` methods